api/src/main/java/net/kyori/adventure/text/JoinConfiguration.java

```diff
@@ -0,0 +1,300 @@
+  /**
+   * Creates a new builder.
+   *
+   * @return a new builder
+   * @since 4.8.0
+   */
+  static @NonNull Builder builder() {
+    return new JoinConfigurationImpl.BuilderImpl();
+  }
+
+  /**
+   * Gets a join configuration with no separators, prefix or suffix.
+   *
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration noSeparators() {
+    return JoinConfigurationImpl.NULL;
+  }
+
+  /**
+   * Creates a join configuration with a separator and no prefix or suffix.
+   *
+   * @param separator the separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separator(final @Nullable ComponentLike separator) {
+    if(separator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).build();
+  }
+
+  /**
+   * Creates a join configuration with a separator and last separator but no prefix or suffix.
+   *
+   * @param separator the separator
+   * @param lastSeparator the last separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separators(final @Nullable ComponentLike separator, final @Nullable ComponentLike lastSeparator) {
+    if(separator == null && lastSeparator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).lastSeparator(lastSeparator).build();
+  }
+
+  /**
+   * Gets the prefix of this join configuration.
+   *
+   * @return the prefix
+   * @since 4.8.0
+   */
+  @Nullable Component prefix();
+
+  /**
+   * Gets the suffix of this join configuration.
+   *
+   * @return the suffix
+   * @since 4.8.0
+   */
+  @Nullable Component suffix();
+
+  /**
+   * Gets the separator of this join configuration.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component separator();
+
+  /**
+   * Gets the last separator of this join configuration.
+   *
+   * @return the last separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparator();
+
+  /**
+   * Gets the operator of this join configuration.
+   *
+   * @return the operator
+   * @since 4.8.0
+   */
+  @NotNull UnaryOperator<Component> operator();
+
+  /**
+   * Gets the last separator that will be used instead of the normal last separator in the case where there
+   * are more than two components being joined. This can be used to mimic a serial (or Oxford) comma.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparatorIfSerial();
+
+  /**
+   * A builder for join configurations.
+   *
+   * @since 4.8.0
+   */
+  interface Builder extends Buildable.Builder<JoinConfiguration> {
+    /**
+     * Sets the prefix of this join configuration builder.
+     *
+     * @param prefix the prefix
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder prefix(final @Nullable Component prefix);
```

Comment:
I don't think we need this method (or others that accept `Component`) in here - `ComponentLike` is good enough, just do the converting to `Component` in the implementation.